src/algebra/monoid_algebra/basic.lean

```diff
@@ -244,11 +244,14 @@ instance [ring k] [semigroup G] : non_unital_ring (monoid_algebra k G) :=
   .. monoid_algebra.non_unital_semiring }
 
 instance [ring k] [mul_one_class G] : non_assoc_ring (monoid_algebra k G) :=
-{ .. monoid_algebra.add_comm_group,
+{ int_cast                    := λ z, single 1 (z : k),
```

Comment:
Possibly out of scope, but it would be good to have the lemma that says
```lean
lemma single_int_cast : single 1 (z : k) = z := rfl
```
and similarly for `nat_cast` (so four such lemmas in total

Reply:
Or actually, `int_cast_def : \u z = single 1 (z : k) := rfl` is probably better, to match `one_def`.